### PR TITLE
Implement Invoices.RefundVoidOpenAmount

### PR DESCRIPTION
**Pull request by katejefferson**
This PR adds implementation for `POST Refund/Void Open Amount` with tests.

### CODE REVIEW

File: mock/services.go [hunk 1]
```diff
@@ -309,6 +312,11 @@ func (m *InvoicesService) MarkFailed(invoiceNumber int) (*recurly.Response, *rec
 	return m.OnMarkFailed(invoiceNumber)
 }
 
+func (m *InvoicesService) RefundVoidOpenAmount(invoiceNumber int, amountInCents int, refundApplyOrder string) (*Response, *Invoice, error) {
+	m.RefundVoidOpenAmountInvoked = true,
```
Comment (cristiangraz) [inline mock/services.go]: Could you remove this comma?

File: invoices_service.go [hunk 2]
```diff
@@ -160,3 +160,23 @@ func (s *invoicesImpl) MarkFailed(invoiceNumber int) (*Response, *Invoice, error
 
 	return resp, &dst, err
 }
+
+// Allows custom invoice amounts to be refunded and generates a refund invoice.
```
Comment (cristiangraz) [inline invoices_service.go]: Could you start this comment with `RefundVoidOpenAmount allows ...`?

File: invoices_service.go [hunk 3]
```diff
@@ -160,3 +160,23 @@ func (s *invoicesImpl) MarkFailed(invoiceNumber int) (*Response, *Invoice, error
...
+// Full open amount refunds of invoices with an unsettled transaction will void
+// the transaction and generate a void invoice.
+// https://dev.recurly.com/docs/line-item-refunds
+func (s *invoicesImpl) RefundVoidOpenAmount(invoiceNumber int, amountInCents int, refundApplyOrder string) (*Response, *Invoice, error) {
+	action := fmt.Sprintf("invoices/%d/refund", invoiceNumber)
+	req, err := s.client.newRequest("POST", action, nil, Invoice{
+		AmountInCents:    amountInCents,
+		RefundApplyOrder: refundApplyOrder,
+	})
```
Comment (cristiangraz) [inline invoices_service.go]: If these two fields are only used to post this one specific endpoint, I'd prefer to create an inline struct and populate the values so we don't clutter the main `Invoice` struct.

```go
data := struct{
   XMLName xml.Name `xml:"invoice"`
   AmountInCents int `xml:"amount_in_cents,omitempty"`
   RefundApplyOrder string `xml:"refund_apply_order,omitempty"`
}{
   AmountInCents: amountInCents,
   RefundApplyOrder: refundApplyOrder,
}
req, err := s.client.newRequest("POST", action, nil, data)
```
Also, if both fields are empty, is recurly okay with receiving `<invoice></invoice>` or should we send no request body in that case? I know you have the sandbox API setup, may need to run some tests (I tried looking through their docs but it's not clear)

File: invoices_test.go [hunk 4]
```diff
@@ -686,3 +686,50 @@ func TestInvoices_MarkFailed(t *testing.T) {
...
+		if !bytes.Equal(b, []byte("<invoice><amount_in_cents>100</amount_in_cents></invoice>")) {
+			t.Fatalf("unexpected input: %s", string(b))
+		}
+		w.WriteHeader(201)
+		fmt.Fprint(w, `<?xml version="1.0" encoding="UTF-8"?><invoice></invoice>`)
+	})
+
+	// Fields ordered in same order as struct xml tags, XML above in same order
+	// for equality check.
+	resp, _, err := client.Invoices.RefundVoidOpenAmount(1010, 100, "")
```
Comment (cristiangraz) [inline invoices_test.go]: Can you include `refund_apply_order` in this test or in a separate test so we can confirm that param is properly sent?

File: mock/services.go (hunk 1 above)
Comment (katejefferson) [inline mock/services.go]: Removed in [e9b88a](https://github.com/blacklightcms/recurly/pull/11/commits/1b52bfba87d06be765b03b13c108236146e9b88a). I actually think the problem was that `*Response` > `*recurly.Response`.

File: invoices_service.go (hunk 2 above)
Comment (katejefferson) [inline invoices_service.go]: Updated in [e9b88a](https://github.com/blacklightcms/recurly/pull/11/commits/1b52bfba87d06be765b03b13c108236146e9b88a).

File: invoices_service.go (hunk 3 above)
Comment (katejefferson) [inline invoices_service.go]: Updated in [e9b88a](https://github.com/blacklightcms/recurly/pull/11/commits/1b52bfba87d06be765b03b13c108236146e9b88a). Also made a comment about what's required for this call.

File: invoices_test.go (hunk 4 above)
Comment (katejefferson) [inline invoices_test.go]: Added in [e9b88a](https://github.com/blacklightcms/recurly/pull/11/commits/1b52bfba87d06be765b03b13c108236146e9b88a).

**Pull request merged**